src/main/java/uk/gov/pay/directdebit/common/model/SearchResponse.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.pay.directdebit.common.model;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.gov.pay.directdebit.payments.model.LinksForSearchResult;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.lang.String.format;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+public class SearchResponse<T> {
+
+    @JsonIgnore
+    private final String gatewayExternalId;
+    @JsonProperty("total")
+    private final Long total;
+    @JsonProperty("count")
+    private final int count;
+    @JsonProperty("page")
+    private final Long page;
+    @JsonProperty("results")
+    private final List<T> results;
+    @JsonProperty("_links")
+    private final LinksForSearchResult linksForSearchResult;
+
+    public SearchResponse(String gatewayExternalId, Long total, Long page, List<T> results, LinksForSearchResult linksForSearchResult) {
+        this.gatewayExternalId = gatewayExternalId;
+        this.total = total;
+        this.count = results.size();
+        this.page = page;
+        this.results = results;
+        this.linksForSearchResult = linksForSearchResult;
+    }
+
+    public List<T> getResults() {
+        return results;
+    }
+
+    public Long getTotal() { return total; }
+
+    public int getCount() { return count; }
+
+    public Long getPage() { return page; }
+
+    public LinksForSearchResult getLinksForSearchResult() { return linksForSearchResult; }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        SearchResponse that = (SearchResponse) o;
+        return Objects.equals(gatewayExternalId, that.gatewayExternalId) &&
+                Objects.equals(total, that.total) &&
+                Objects.equals(page, that.page);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(gatewayExternalId, total, page);
+    }
```

Comment:
Should all the fields be in the `equals(…)` and `hashCode()` methods?

Reply:
Fair point, I've regened the methods.